developers/compiling.rst

```diff
@@ -10,34 +10,182 @@ Compiling Dash Core
-and only if they match, the build is accepted and uploaded to dash.org.
-
-Instructions on how to build Dash Core 0.13.0 will appear here once the
-Docker build system for Gitian is available. Instructions to create
-deterministic builds of Dash Core 0.12.3 or older are available `here
-<https://docs.dash.org/en/0.12.3/developers/compiling.html#gitian-build>`__ 
-on a previous version of this page.
+Gitian is the deterministic build process that is used to build the Dash Core
+executables. It provides a way to be reasonably sure that the executables are
+really built from the source on GitHub. It also makes sure that the same, tested
+dependencies are used and statically built into the executable. Multiple
+developers build the source code by following a specific descriptor ("recipe"),
+cryptographically sign the result, and upload the resulting signature. These
+results are compared and only if they match, the build is accepted and uploaded
+to dash.org.
+
+Build process
+=============
+
+This setup has been tested using a clean install of Ubuntu 20.04. Start by
+logging in as the "root" user. Create a new user with the following command,
+replacing ``<username>`` with a username of your choice::
+
+  adduser <username>
+
+You will be prompted for a password. Enter and confirm using a new password
+(different to your root password) and store it in a safe place. You will also
+see prompts for user information, but this can be left blank. Alternatively, an
+existing user can be used on systems that are already in use (e.g. your existing
+development system).
+
+Add the user to the sudo group so they can perform commands as root::
+
+  usermod -aG sudo <username>
+
+Install prerequisites
+---------------------
+
+While still logged in as root, update the system from the Ubuntu package
+repository::
+
+  apt update
+  apt upgrade -y
+
+Install apt-cacher-ng::
+
+  apt install -y apt-cacher-ng
+
+.. note::
+  Select ``No`` when asked ``Allow HTTP tunnels through Apt-Cacher NG?`` during
+  installation.
+
+  Note: you may also need to open port 3142 if you have a firewall enabled on
+  your system (e.g. ``ufw allow 3142/tcp``).
+
+After installing these updates, reboot the system, login as ``<username>``, and
+clone required repositories::
+
+  git clone https://github.com/dashpay/dash
+  git clone https://github.com/devrandom/gitian-builder
+  git clone https://github.com/dashpay/dash-detached-sigs
+  git clone https://github.com/dashpay/gitian.sigs
+
+Download the Mac OSX SDK::
+
+  mkdir gitian-builder/inputs
+  wget -q -O gitian-builder/inputs/MacOSX10.11.sdk.tar.gz https://bitcoincore.org/depends-sources/sdks/MacOSX10.11.sdk.tar.gz
+
+Prepare gitian
+--------------
+  
+It is only necessary to run this step during the initial setup of your machine::
+
+  # <signer> = The name associated with your PGP key
+  # <version> = Dash Core tag to build
+  ./dash/contrib/gitian-build.py --setup <signer> <version>
```

Comment:
Examples would be great here, I ran through this first and included the `v` in the <version> tag, which broke things later with `vv0.17.0.3`.

Reply:
Updated with example cmds